plugins/woocommerce/src/Admin/Features/ProductBlockEditor/ProductTemplates/GroupInterface.php

```diff
@@ -16,4 +17,11 @@ interface GroupInterface extends BlockContainerInterface {
 	 * @return SectionInterface new block section.
 	 */
 	public function add_section( array $block_config ): SectionInterface;
+
+	/**
+	 * Adds a new block to the section block.
+	 *
+	 * @param array $block_config block config.
+	 */
+	public function add_block( array $block_config ): BlockInterface;
```

Comment:
Is there some other discussion on reverting to `add_block` being allowed on these blocks?  Group blocks should not have the ability to use the `add_block` method.

I'm seeing that `Group` is extended from `ProductBlock` which has a public `get_block` method and this also feels incorrect to me.

If this block is meant to be guarded and only allow additions of sections, we should not expose this as a public method.

Reply:
cc @louwie17 in case there was some other discussion on this decision that I'm not aware of.

Reply:
We are adding blocks directly to groups in SImpleProductTemplate.

For example, a notice is added to the pricing group...

https://github.com/woocommerce/woocommerce/blob/7da226ad63ce80440eb3e17778304296bee63c58/plugins/woocommerce/src/Admin/Features/ProductBlockEditor/ProductTemplates/SimpleProductTemplate.php#L324-L335

Reply:
(I added this to the interface after noticing that VSCode was warning me about calling an unknown method `add_block()` on the group)

Reply:
Thanks for the notices example; I was not aware that we were adding non-sections to blocks.

That's unfortunate since this prevents our ability to adequately guard these parent blocks.

Is there a predefined list of blocks we can say are allowed to be added to groups?  For example, sections, notices, etc?  Or are there too many possibilities and edge cases to cover?

Reply:
> Is there a predefined list of blocks we can say are allowed to be added to groups? For example, sections, notices, etc? Or are there too many possibilities and edge cases to cover?

We could make things a bit more generic and have a general "notice" block type that we allow. And maybe something else. But, I worry that any limitation we put in place could prevent a third-party from doing something completely legit that we just haven't thought of.

I think that guarding the top level to only allow for the adding of groups is the most important, because we really have no UI concept for how anything other than a group (tab) would be shown in that top level. But, underneath groups, I think the UI can handle "anything" being put there (though, certainly things could get messy).

All that said, if we wanted to lock it down more initially, we could add a `add_notice()` to GroupInterface, and then only consider opening it up more if we got feedback from third-parties (or ourselves) that we needed more flexibility. Note that unless we also put some more constraints in place, that would still allow for notices to be places anywhere within a group. For example, it would allow the following:

* Group
    * Section A
    * Notice
    * Section B

Which seems odd, but maybe legit in some scenario.

I'm in favor of providing light guardrails in the API, but ultimately letting extensions do just about anything. Our documentation and UI guidelines can provide best practices, but there is usually a reasonable case to ignore a guideline eventually.

Reply:
>I'm in favor of providing light guardrails in the API, but ultimately letting extensions do just about anything. Our documentation and UI guidelines can provide best practices, but there is usually a reasonable case to ignore a guideline eventually.

I agree with this and I'm sure if we add the notice method, we'll eventually end up with quite a few more methods to cover various use cases.

Let's leave it open.